### PR TITLE
Remove some xtensor from refinement code.

### PR DESCRIPTION
**Pull request by jorgensd**
- Use plain loops to copy. 
- Keep midpoint structure flat
- Reduce size of new_geometry from the start, instead of doing a view copy at the end.
- Remove comment about things that should work with icpx

### CODE REVIEW

File: cpp/dolfinx/refinement/utils.cpp [hunk 1]
```diff
@@ -86,16 +86,16 @@ xt::xtensor<double, 2> create_new_geometry(
...
   const std::size_t num_new_vertices = local_edge_to_new_vertex.size();
   xt::xtensor<double, 2> new_vertex_coordinates(
-      {num_vertices + num_new_vertices, 3});
+      {num_vertices + num_new_vertices, gdim});
 
   for (std::size_t v = 0; v < num_vertices; ++v)
   {
     const int pos = 3 * vertex_to_x[v];
-    for (std::size_t j = 0; j < 3; ++j)
+    for (std::size_t j = 0; j < gdim; ++j)
```
Comment (garth-wells) [inline cpp/dolfinx/refinement/utils.cpp:98]: It's better for performance to have a fixed loop bound here.

Comment (jorgensd) [reply cpp/dolfinx/refinement/utils.cpp:98]: Even if it means that we need to do a strided copy when returning the data?

Comment (garth-wells) [reply cpp/dolfinx/refinement/utils.cpp:98]: Make everything 3D.

Comment (jorgensd) [reply cpp/dolfinx/refinement/utils.cpp:98]: We need to remove the last dimensions before passing this to the geometry constructor, as it computes the gdim from the shape of the input.

**Review by garth-wells — APPROVED** [review]

**Pull request merged**